concourse/steps/update_component_deps.py

```diff
@@ -410,6 +572,32 @@ def create_upgrade_pr(
     )
     from_component = from_component_descriptor.component
 
+    to_component_descriptor = component_descriptor_lookup(
+        ocm.ComponentIdentity(
+            name=to_ref.componentName,
+            version=to_version,
+        )
+    )
+
+    to_component = to_component_descriptor.component
+
+    cfg_factory = ctx.cfg_factory()
```

Comment:
do not create cfg-factories within nested functions. instead, pass-in a cfg-factory into the function, if needed. or better yet: pass-in the needed bom_diff-object